fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/db2/Db2Adapter.java

```diff
@@ -365,43 +365,83 @@ public void detachPartition(String schemaName, String tableName, String partitio
-                String qualifiedName = DataDefinitionUtil.getQualifiedName(schemaName, t.getName());
+                String qualifiedName = DataDefinitionUtil.getQualifiedName(schemaName, t.getObjectName());
                 throw new DataAccessException("No partition information found for table: " + qualifiedName);
             }
             else {
                 final String partitionName = "TENANT" + tenantId;
-                final String targetTableName = DataDefinitionUtil.getQualifiedName(schemaName, t.getName() + "_" + partitionName);
-                removeTenantPartition(schemaName, t.getName(), partitionName, targetTableName, tenantStagingTable);
+                final String targetTableName = getDetachedPartitionTableName(t, tenantId);
+                detachPartition(schemaName, t.getObjectName(), partitionName, targetTableName);
             }
         }
     }
 
-    protected void removeTenantPartition(String schemaName, String tableName, String partitionName, String targetTableName,
-            String tenantStagingTable) {
-
-        // Detach the given partition of the table into the targetTableName (within the same schema).
-        detachPartition(schemaName, tableName, partitionName, targetTableName);
+    @Override
+    public void dropDetachedPartitions(Collection<Table> tables, String schemaName, int tenantId) {
 
-        // We need to add this target table name to the tenantStagingTable so that we
-        // can remember to delete it later...an operation which can only be done in
-        // a new transaction
-        Db2AddTableToStaging cmd = new Db2AddTableToStaging(schemaName, tenantStagingTable, targetTableName);
-        runStatement(cmd);
+        // Only process tables which are partitioned
+        Map<String, PartitionInfo> partitionInfoMap = new HashMap<>();
+        loadPartitionInfoMap(partitionInfoMap, schemaName);
+        
+        for (Table t : tables) {
+            PartitionInfo pi = partitionInfoMap.get(t.getObjectName());
+            if (pi == null) {
+                // We should only be dealing with partitioned tables at this stage, so this
+                // is a fatal error
+                String qualifiedName = DataDefinitionUtil.getQualifiedName(schemaName, t.getObjectName());
+                throw new DataAccessException("No partition information found for table: " + qualifiedName);
+            } 
+            else {
+                // drop the table which now represents the detached partition
+                final String detachedPartitionTableName = getDetachedPartitionTableName(t, tenantId);
+                try {
+                    logger.info("Dropping detached partition (table): '" + detachedPartitionTableName + "'");
+                    dropTable(schemaName, detachedPartitionTableName);
+                } 
+                catch (Exception x) {
+                    // we want this to be idempotent, so we suppress propagation of any error
```

Comment:
format, could you more "catch (Exception x) {" to above line?

Reply:
done